### PR TITLE
[DOCS] add content to on_k8s.md

### PR DESCRIPTION
**Pull request by zwangsheng**
zwangsheng [Closes #472](https://github.com/yaooqinn/kyuubi/pull/472)    Target Issue [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Add content to on_k8s.md about deploying kyuubi on k8s

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/deployment/on_kubernetes.md [hunk 1]
```diff
@@ -5,3 +5,97 @@
...
+* Setup Hadoop client configurations at the machine which Kyuubi server locates
+
+## Configurations
+
+### Environment
+
+Either `HADOOP_CONF_DIR` is configured and points to the Hadoop client configurations directory,
+usually, `$HADOOP_HOME/etc/hadoop`.
+
+If the `HADOOP_CONF_DIR` points the K8S and HDFS cluster correctly, you should be able to run the `SparkPi` example on
```
Comment (yaooqinn) [inline docs/deployment/on_kubernetes.md:27]: A Pi job have nothing to do with hdfs with k8s

File: docs/deployment/on_kubernetes.md [hunk 2]
```diff
@@ -5,3 +5,97 @@
...
+* A binary distribution of Spark which is built with K8S
+* An active K8S cluster
+* An active Apache Hadoop HDFS cluster
+* Setup Hadoop client configurations at the machine which Kyuubi server locates
+
+## Configurations
+
+### Environment
+
+Either `HADOOP_CONF_DIR` is configured and points to the Hadoop client configurations directory,
```
Comment (yaooqinn) [inline docs/deployment/on_kubernetes.md:24]: this seems to be only needed when accessing hdfs

File: docs/deployment/on_kubernetes.md (hunk 1 above)
Comment (zwangsheng) [reply docs/deployment/on_kubernetes.md:27]: so, wo should provide another example, or just remove this `HADOOP_CONF_DIR`

File: docs/deployment/on_kubernetes.md [hunk 3]
```diff
@@ -5,3 +5,97 @@
 </div>
 
 # Running Kyuubi on Kubernetes
+
+## Requirements
+
+When you want to deploy Kyuubi’s Spark SQL engines on Kubernetes(use K8S as the name below), you’d better have cognition
```
Comment (ulysses-you) [inline docs/deployment/on_kubernetes.md:11]: Deploy engine or deploy Kyuubi ? I guess we should support both, and we need document these.

But if this doc is about deply engine, it's better to specify at title and description.

**Comment by zwangsheng** [comment]
@yaooqinn how about change to this

File: docs/deployment/on_kubernetes.md (hunk 3 above)
Comment (zwangsheng) [reply docs/deployment/on_kubernetes.md:11]: Maybe, i can add some about deploy kyuubi in this.

File: docs/deployment/on_kubernetes.md [hunk 4]
```diff
@@ -5,3 +5,87 @@
...
```
Comment (pan3793) [inline docs/deployment/on_kubernetes.md:11]: > Kubernetes, also known as K8s, ...

Please observe the character case, use `K8s` instead of `K8S`

**Comment by zwangsheng** [comment]
will open new PR for Kubernetes DOCS
